waku/v2/peermanager/connection_gater_test.go

```diff
@@ -0,0 +1,86 @@
+package peermanager
+
+import (
+	"context"
+	"github.com/libp2p/go-libp2p/core/control"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peerstore"
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+	"github.com/stretchr/testify/require"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+	"testing"
+)
+
+type mockConnMultiaddrs struct {
+	local, remote ma.Multiaddr
+}
+
+func (m mockConnMultiaddrs) LocalMultiaddr() ma.Multiaddr {
+	return m.local
+}
+
+func (m mockConnMultiaddrs) RemoteMultiaddr() ma.Multiaddr {
+	return m.remote
+}
+
+func TestConnectionGater(t *testing.T) {
+
+	log := utils.Logger()
+
+	_, h1, _ := makeWakuRelay(t, log)
+	_, h2, _ := makeWakuRelay(t, log)
+
+	h1.Peerstore().AddAddrs(h2.ID(), h2.Addrs(), peerstore.PermanentAddrTTL)
+	err := h1.Connect(context.Background(), h2.Peerstore().PeerInfo(h2.ID()))
+	require.NoError(t, err)
+
+	h2.Peerstore().AddAddrs(h1.ID(), h1.Addrs(), peerstore.PermanentAddrTTL)
+	err = h2.Connect(context.Background(), h1.Peerstore().PeerInfo(h1.ID()))
+	require.NoError(t, err)
+
+	peerA := h1.ID()
+
+	remoteAddr1 := ma.StringCast("/ip4/1.2.3.4/tcp/1234")
+
+	connGater := NewConnectionGater(2, log)
+
+	// Test peer blocking
+	allow := connGater.InterceptPeerDial(peerA)
```

Comment:
Did not understand the need for this, as the function InterceptPeerDial doesn't do anything but just returns.

Infact many of these intercept* functions don't have any implementation other than a blind return, so no point in writing tests for them

Reply:
These functions are tested for consistency. It is not possible to cover all functionality at once. When I will return back either because of a problem, or because I want to add a test, I will be able to quickly see where are untested spots. More spots -> more complicated to asses what to do about them. I may not remember connection gater had so simple implementation.